driver-core/src/main/com/mongodb/ServerApi.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import com.mongodb.annotations.NotThreadSafe;
+import com.mongodb.lang.Nullable;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * A specification of the server API on which the application relies.
+ *
+ * @since 4.3
+ */
+public final class ServerApi {
+
+    private final ServerApiVersion version;
+    private final Boolean deprecationErrors;
+    private final Boolean strict;
+
+    private ServerApi(final ServerApiVersion version, final @Nullable Boolean strict, final @Nullable Boolean deprecationErrors) {
+        this.version = notNull("version", version);
+        this.deprecationErrors = deprecationErrors;
+        this.strict = strict;
+    }
+
+    /**
+     * Gets the server API version
+     *
+     * @return the server API version
+     */
+    public ServerApiVersion getVersion() {
+        return version;
+    }
+
+    /**
+     * Gets whether the application requires strict server API version enforcement.
+     *
+     * <p>
+     * The default is false.
+     * </p>
+     *
+     * @return whether the application requires strict server API version enforcement
```

Comment:
Is it ok not to mention its an optional?

Reply:
I think it's ok.  The type of the return tell you that, which is convenient.

Reply:
Ack